packages/next/client/components/react-dev-overlay/hot-reloader-client.tsx

```diff
@@ -176,16 +176,20 @@ function tryApplyUpdates(
   module.hot
     .check(/* autoApply */ false)
     .then((updatedModules: any[] | null) => {
-      const hasUpdates = Boolean(updatedModules?.length)
+      if (!updatedModules) {
+        return null
+      }
+
       if (typeof onBeforeUpdate === 'function') {
+        const hasUpdates = Boolean(updatedModules.length)
         onBeforeUpdate(hasUpdates)
```

Comment:
FWIW, `hasUpdates` may not be necessary. Seems confusing to handle the `null` and empty array cases differently. I don't have enough webpack knowledge to know if we should still call `.apply()` in the latter case, but that does appear to be what webpack does when `autoApply=true`.